exercises/zebra-puzzle/zebra_puzzle_test.go

```diff
@@ -1,23 +1,22 @@
-package zebra_puzzle
+package zebra
 
 import "testing"
 
-func TestSolve(t *testing.T) {
-	const (
-		expectedWaterDrinker = "Norwegian"
-		expectedZebraOwner   = "Japanese"
-	)
-	waterDrinker, zebraOwner := Solve()
-	if waterDrinker != expectedWaterDrinker {
-		t.Fatalf("FAILED: For waterDrinker, expected %q, actual %q", expectedWaterDrinker, waterDrinker)
+func TestSolvePuzzle(t *testing.T) {
+	expected := Solution{DrinksWater: "Norwegian", OwnsZebra: "Japanese"}
+	actual := SolvePuzzle()
+	if expected.DrinksWater != actual.DrinksWater {
```

Comment:
What about checking structs directly and pretty printing them if differs ?

Reply:
That's true.  But then it would still have the two tests that are there now in order to print the specific error; that seemed a bit redundant,  so I omitted the struct compare.

Reply:
Oh, I get it. Just use a print that prints both.